core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -657,6 +681,28 @@ class BlockManagerMasterEndpoint(
     }
   }
 
+  private def getShufflePushMergerLocations(
+      numMergersNeeded: Int,
+      hostsToFilter: Set[String]): Seq[BlockManagerId] = {
+    val activeBlockManagers = blockManagerIdByExecutor.groupBy(_._2.host)
+      .mapValues(_.head).values.map(_._2).toSet
+    val filteredActiveBlockManagers = activeBlockManagers
+      .filterNot(x => hostsToFilter.contains(x.host))
+    val filteredActiveMergers = filteredActiveBlockManagers.map(
+      x => BlockManagerId(x.executorId, x.host, StorageUtils.externalShuffleServicePort(conf)))
+
+    // Enough mergers are available as part of active executors list
+    if (filteredActiveMergers.size >= numMergersNeeded) {
+      filteredActiveMergers.toSeq
+    } else {
+      // Delta mergers added from inactive mergers list to the active mergers list
+      val filteredDeadMergers = shuffleMergerLocations.values
+        .filterNot(mergerHost => filteredActiveMergers.exists(x => x.host == mergerHost.host))
+      filteredActiveMergers.toSeq ++
+        filteredDeadMergers.toSeq.take(numMergersNeeded - filteredActiveMergers.size)
```

Comment:
This will mean we will keep prioritizing hosts which will be removed in `addMergerLocation` when thresholds are hit.
Is this intentional ?

Reply:
Yes, though this is not intentional. It seems like a side-effect. Do you think we should prefer `active executors` locations which are also part of `shuffleMergerLocations` before other `active executors` locations?

Reply:
We are already prioritizing hosts with active executors - this is making up the remaining in case number of merger hosts with active executors is not sufficient.
I was trying to understand the behavior ... random order might be better instead of preferring the head or tail of this map. Thoughts @tgravescs ?

Reply:
+1 for random order.